src/EnergyPlus/DaylightingManager.cc

```diff
@@ -566,75 +559,61 @@ void CalcDayltgCoefficients(EnergyPlusData &state)
                 print(state.files.eio, Format_700);
-                for (ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
-                    if (state.dataDaylightingData->ZoneDaylight(ZoneNum).NumOfDayltgExtWins == 0 ||
-                        state.dataDaylightingData->ZoneDaylight(ZoneNum).DaylightMethod != DataDaylighting::iDaylightingMethod::SplitFluxDaylighting)
-                        continue;
-                    for (loop = 1; loop <= state.dataDaylightingData->ZoneDaylight(ZoneNum).NumOfDayltgExtWins; ++loop) {
-                        IWin = state.dataDaylightingData->ZoneDaylight(ZoneNum).DayltgExtWinSurfNums(loop);
-                        // For this report, do not include ext wins in zone adjacent to ZoneNum since the inter-reflected
-                        // component will not be calculated for these windows until the time-step loop.
-                        if (state.dataSurface->Surface(IWin).SolarEnclIndex == Zone(ZoneNum).zoneFirstSpaceSolEnclosure) {
-                            // Output for each reference point, for each sky. Group by sky type first
-                            for (const DataDaylighting::SkyType &skyType : {DataDaylighting::SkyType::Clear,
-                                                                            DataDaylighting::SkyType::ClearTurbid,
-                                                                            DataDaylighting::SkyType::Intermediate,
-                                                                            DataDaylighting::SkyType::Overcast}) {
-                                std::string skyTypeString;
-                                if (skyType == DataDaylighting::SkyType::Clear) {
-                                    skyTypeString = "Clear Sky";
-                                } else if (skyType == DataDaylighting::SkyType::ClearTurbid) {
-                                    skyTypeString = "Clear Turbid Sky";
-                                } else if (skyType == DataDaylighting::SkyType::Intermediate) {
-                                    skyTypeString = "Intermediate Sky";
-                                } else if (skyType == DataDaylighting::SkyType::Overcast) {
-                                    skyTypeString = "Overcast Sky";
-                                    //} else {
-                                    //    // Should never happen
-                                    //    skyTypeString = "ERROR_SKY_TYPE_NOT_HANDLED";
-                                }
+                for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
+                    for (int controlNum = 1; controlNum <= state.dataDaylightingData->totDaylightingControls; ++controlNum) {
+                        auto &thisDaylightControl = state.dataDaylightingData->daylightControl(controlNum);
+                        if (thisDaylightControl.zoneIndex != zoneNum) continue;
```

Comment:
These types of checks suggest that there is a more efficient way of iterating over the set you are trying to iterate over.

Reply:
Yes, this is here just to maintain order for now.
`                // TODO MJW: Post-release change to enclosure name and remove zone loop (which is here to preserve order for now)`